packages/klingon-ui/src/app/_shared/terminal/terminal.service.ts

```diff
@@ -100,6 +100,10 @@ export class TerminalService {
     return this.cli.runNgCommand(data);
   }
 
+  command(data) {
```

Comment:
I think server side validation would be better. We can allow only certain commands from server side. So, no matter what commands user run from client, server will run only allowed ones. @manekinekko

Reply:
We can take reference from OWASP:- [OS Command Injection Defense Cheat Sheet](https://www.owasp.org/index.php/OS_Command_Injection_Defense_Cheat_Sheet)

Reply:
1) escapeshellarg adds single quotes around a string and quotes/escapes any existing single quotes allowing you to pass a string directly to a shell function and having it be treated as a single safe argument.
http://locutus.io/php/exec/escapeshellarg/

2) How To Prevent Command Injection
https://affinity-it-security.com/how-to-prevent-command-injection/